[Feat] Argument Resolver 통한 인증 확인 중복 제거

Bumnote opened the pull request:
## #️⃣ 연관된 이슈

#21 

## 📝 작업 내용

-Argument Resolver 활용하여 회원 인증 정보 확인을 위한 불필요한 중복 코드를 제거하였습니다.

<img width="600" alt="image" src="https://github.com/user-attachments/assets/626de768-7f81-4522-ab10-5af13fdef854" />



<br/>

## 💬 리뷰 요구사항

- Argument Resolver 개념을 잘 구현했는지 검토해주세요!

Bumnote committed:
[Feat] Argument Resolver 통한 인증 확인 중복 제거

[수정 내용]
- Argument Resolver 활용하여 인증 정보 중복 확인 제거
- yml 파일 업로드(env 설정)

arkss commented on src/main/java/catchtable/cooking/config/UserArgumentResolver.java:
```diff
@@ -0,0 +1,45 @@
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class UserArgumentResolver implements HandlerMethodArgumentResolver {
```
얘는 config는 아니죠!
얘는 Resolver를 패키지 만들어서 그 안에 넣읍시다!

Bumnote replied:
[02e0f65](https://github.com/Bumnote/catchTable/pull/23/commits/02e0f6534593a0f1363471054df4b816e8e032a9)

해당 커밋에서 반영했습니다!

arkss commented on src/main/java/catchtable/cooking/config/UserArgumentResolver.java:
```diff
@@ -0,0 +1,45 @@
+        String nickname = jwtTokenProvider.getSubject(token);
+
+        log.info("resolver nickname: {}", nickname);
+        return memberRepository.findByNickname(nickname).orElse(null);
```
null을 반환하는거보다 exception을 반환하는게 맞다고 생각해요!

Bumnote replied:
[02e0f65](https://github.com/Bumnote/catchTable/pull/23/commits/02e0f6534593a0f1363471054df4b816e8e032a9)

해당 커밋에서 적용했습니다 :)

arkss commented on src/main/java/catchtable/cooking/controller/MemberController.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @GetMapping("/api/test")
+    public CommonResponse<?> test(@UserArg Member member) {
```
@UserArg 얘도 없으면 더 좋을 거 같아요!
이것까지만 해보시죠!!

Bumnote replied:
[02e0f65](https://github.com/Bumnote/catchTable/pull/23/commits/02e0f6534593a0f1363471054df4b816e8e032a9) 

해당 커밋에서 적용했습니다!

Bumnote committed:
[Refactor] Argument Resolver 기능 수정

[수정 내용]
- Controller 매개변수로 @UserArg 없이도 Member 인식
- Argument Resolver 단에서 예외처리
- Resolver 폴더를 만들어서 Resolver 파일 저장

arkss commented on src/main/java/catchtable/cooking/resolver/UserArgumentResolver.java:
```diff
@@ -39,7 +40,8 @@ public Member resolveArgument(MethodParameter parameter,
-        log.info("resolver nickname: {}", nickname);
-        return memberRepository.findByNickname(nickname).orElse(null);
+        return memberRepository.findByNickname(nickname).orElseThrow(
+                () -> new CustomException(Code.NOT_EXIST_NICKNAME)
```
jwt에 nickname을 넣는게 아니라 member 객체를 (or 필요한 정보만 담긴 member DTO를) 저장하면 여기서 DB 조회를 줄일 수 있을거예요!

Bumnote replied:
[b1df105](https://github.com/Bumnote/catchTable/pull/28/commits/b1df105bfe86cd9476aad74af87f543d252e6d25)

해당 커밋에서 반영했습니다! 확실히 JWT 토큰에 저장된 객체 내용을 가지고, 이곳 저곳 활용함으로써 DB 접근을 최소화할 수 있었습니다 :) 😊
